tests/components/apple_tv/common.py

```diff
@@ -0,0 +1,138 @@
+"""Test code shared between test files."""
+
+from functools import partial
+
+from pyatv import conf, interface
+from pyatv.const import Protocol
+
+from homeassistant.data_entry_flow import AbortFlow
+
+
+class MockPairingHandler(interface.PairingHandler):
+    """Mock for PairingHandler in pyatv."""
+
+    def __init__(self, *args):
+        """Initialize a new MockPairingHandler."""
+        super().__init__(*args)
+        self.always_fail = False
+
+    def pin(self, pin):
+        """Pin code used for pairing."""
+        self.pin_code = pin
+        self.paired = False
+
+    @property
+    def device_provides_pin(self):
+        """Return True if remote device presents PIN code, else False."""
+        return self.service.protocol in [Protocol.MRP, Protocol.AirPlay]
+
+    @property
+    def has_paired(self):
+        """If a successful pairing has been performed.
+
+        The value will be reset when stop() is called.
+        """
+        return not self.always_fail and self.paired
+
+    async def begin(self):
+        """Start pairing process."""
+
+    async def finish(self):
+        """Stop pairing process."""
+        self.paired = True
+        self.service.credentials = self.service.protocol.name.lower() + "_creds"
+
+
+class FlowInteraction:
```

Comment:
Ok, I added this along the way because of the same reason more or less. They became very long, repetitive and hard to follow and maintain. I can remove it but it's gonna take a *lot* of work.